kernel/src/component.rs

```diff
@@ -0,0 +1,30 @@
+/// A Component extends the functionality of the Tock kernel.
+/// This abstraction is intended to make the kernel boot sequence simpler:
+/// without it, the reset_handler involves lots of driver-specific
+/// initialization. The Component trait encapsulates all of the
+/// initialization and configuration of a kernel extension inside
+/// the finalize function call.
+///
+/// Note that instantiating a component does not necessarily instantiate
+/// the underlying Output type; this can be instantiated when finalize()
+/// is called.
+
+pub trait Component {
+    type Output;
```

Comment:
Document what this is expected to be and how it will/could be used. I believe generally it should be a `&'static`, right?